SANC-32-Implement-passenger-management-system

wesleylui opened the pull request:
Add passenger management schema with contact information.
Passengers table includes:
- name, email (unique), phone
- notes (accessibility needs or special requests)
- emergency contact info (name, phone, relation)
- createdAt and updatedAt timestamps

Also did:
- Booking-Passenger join table to associate passengers with bookings
- add ```passenger-schema``` to ```drizzle.config.ts```

The /drizzle SQL files:
- ```0000_long_ikaris.sql``` needed for my local supabase instance
- ```0000_tough_drax.sql``` contains passenger schema sql

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added passenger management functionality to bookings system.

* **Bug Fixes**
  * Streamlined user authentication flow and removed deprecated email service.

* **Refactoring**
  * Removed organization/team and invitation management system.
  * Removed calendar view component.
  * Removed login and account registration pages.
  * Updated user role system (admin, driver, agency).

* **Chores**
  * Updated dependencies and build configurations.
  * Simplified environment configuration.
  * Updated code organization standards.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jason-duong4509 committed:
SANC-39: Features are coded in. Likely will rewrite them to fix prevalent bugs

- Direction address validation -> Mostly completed. An error with
  properly validating an address must be fixed
- Link integration -> Must be rewritten as it was coded improperly

jason-duong4509 committed:
SANC-39: Added backend validation for destination address

- Possible use case: frontend calls a DB update function from backend.
  Backend verifies form inputs and calls this new function to validate
  destination address

jason-duong4509 committed:
SANC-39: Wrote frontend validation for destination address

- Not fully tested yet

jason-duong4509 committed:
SANC-39: Completed Task

- \src\app\_components\agencycomponents\agency-interactive-area.tsx
  contains temporary comments on how to call an endpoint that verifies
  the destination location in the backend
- \src\app\page.tsx contains temporary comments that may be used to
  replace link integration at a future date

wesleylui committed:
implement passenger db schema

jason-duong4509 committed:
SANC-39: Updated code to address PR feedback

Made the following changes to address PR feedback:
- Made a separate private API key for Google Maps
- Added error handling when making a fetch call to Google Maps' API and when receiving a response from it
- Removed NPM usage and added yarn usage
- Made the backend validation endpoint for destination address return null instead of false if validation failed
- Updated development API keys (in Google cloud) to ensure that features work with heavily restricted keys

Fixed an edge case arising from filled user input in the mantine form causing Google Maps' autocorrect API to break
- Done by clearing the destination address form input upon closing it

burtonjong commented on src/server/db/passenger-schema.ts:
I think i'm missing something ehre. What was the point of this ticket... is it to assocaite a passenger to a bookign?  if so why do we need the booking passengers talbe..?

wesleylui replied:
Here's what I was thinking...I added the `booking_passengers` table as a join table for many-to-many relationships
For ex: a booking can have multiple passengers, and a passenger can be on multiple bookings. This was to avoid duplicating passenger contact information if a passenger were to be ride multiple times.

Questions:
- should a booking have multiple passengers?
- passengers are **not** users of the app? are they independent passenger records created as part of a booking?
What do you suggest?

jason-duong4509 committed:
Merge remote-tracking branch 'origin/main' into SANC-39-Google-maps-link-integration

# Conflicts:
#	.env.example
#	package.json
#	src/app/_components/agencycomponents/agency-interactive-area.tsx
#	src/app/page.tsx
#	yarn.lock

jason-duong4509 committed:
SANC-39: Updated code to address PR feedback again

- Changed the way the Google Maps API was loaded to follow an automatic
  approach rather than manually loading it
- Utilized env.js to properly handle API keys

jason-duong4509 committed:
Merge pull request #18 from Code-the-Change-YYC/SANC-39-Google-maps-link-integration

SANC-39: Google maps link integration

wesleylui committed:
Merge branch 'main' of https://github.com/Code-the-Change-YYC/salvationarmy into SANC-32-Implement-passenger-management-system

burtonjong commented:
@wesleylui not really sure what happened to this branch... maybe reset off main

promatty commented:
^^ what burton said.. ts blew up fr
